src/lib/rules/no-insecure-cookies/no-insecure-cookies.ts

```diff
@@ -0,0 +1,314 @@
+/**
+ * @fileoverview This rule validates the `set-cookie` header and confirms that it is sent with `Secure` and `HttpOnly` directive over HTTPS.
+ */
+
+import { RuleContext } from '../../rule-context'; // eslint-disable-line no-unused-vars
+import { debug as d } from '../../utils/debug';
+import { IFetchEnd, IAsyncHTMLElement, IResponse, IRuleBuilder, IRule } from '../../types'; // eslint-disable-line no-unused-vars
+import { isHTTPS, isRegularProtocol, normalizeString } from '../../utils/misc';
+
+const debug = d(__filename);
+
+// ------------------------------------------------------------------------------
+// Public
+// ------------------------------------------------------------------------------
+
+const rule: IRuleBuilder = {
+    create(context: RuleContext): IRule {
+        /** A collection of accepted attributes
+         * See https://stackoverflow.com/questions/19792038/what-does-priority-high-mean-in-the-set-cookie-header for details about the `priority` attribute.
+        */
+        const acceptedCookieAttributes: Array<string> = ['expires', 'max-age', 'domain', 'path', 'secure', 'httponly', 'samesite', 'priority'];
+        /** A collection of illegal characters in cookie name
+         * Reference: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#Directives
+        */
+        const illegalCookieNameChars: string = '()<>@,;:\"/[]?={}'; // eslint-disable-line no-useless-escape
+        /** A collection of illegal characters in cookie value
+         * Reference: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#Directives
+         */
+        const illegalCookieValueChars: string = ',;"/';
+        /** Header name used in report */
+        const headerName: string = 'set-cookie';
+
+        /** Object containing set-cookie header information after parsing */
+        type ParsedSetCookie = { // eslint-disable-line no-unused-vars
```

Comment:
If these aren't going to be used anywhere else maybe a types file in the same folder as the rule? Don't know :/